clio-client/src/main/scala/org/broadinstitute/clio/client/commands/Commands.scala

```diff
@@ -1,33 +1,80 @@
 package org.broadinstitute.clio.client.commands
 
-import akka.http.scaladsl.model.HttpResponse
-import enumeratum._
-import org.broadinstitute.clio.client.parser.BaseArgs
-import org.broadinstitute.clio.client.util.IoUtil
-import org.broadinstitute.clio.client.webclient.ClioWebClient
+import java.time.OffsetDateTime
+import java.util.UUID
 
-import scala.collection.immutable.IndexedSeq
-import scala.concurrent.{ExecutionContext, Future}
-
-sealed trait CommandType extends EnumEntry
+import akka.http.scaladsl.model.headers.OAuth2BearerToken
+import caseapp.Recurse
+import enumeratum.{Enum, EnumEntry}
+import org.broadinstitute.clio.transfer.model.{
+  TransferWgsUbamV1Key,
+  TransferWgsUbamV1Metadata,
+  TransferWgsUbamV1QueryInput
+}
+import caseapp._
 
-object Commands extends Enum[CommandType] {
-  override val values: IndexedSeq[CommandType] = findValues
+import scala.reflect.ClassTag
+import scala.util.{Failure, Success, Try}
 
-  case object AddWgsUbam extends CommandType
-  case object QueryWgsUbam extends CommandType
-  case object MoveWgsUbam extends CommandType
+object Commands {
 
-  val pathMatcher = Map(
-    AddWgsUbam.toString -> AddWgsUbam,
-    QueryWgsUbam.toString -> QueryWgsUbam,
-    MoveWgsUbam.toString -> MoveWgsUbam
+  final case class CommonOptions(
+    bearerToken: OAuth2BearerToken = OAuth2BearerToken("")
   )
 
-}
+  implicit def enumEntryParser[T <: EnumEntry: Enum](
+    implicit c: ClassTag[T]
+  ): core.ArgParser[T] = {
+    core.ArgParser.instance[T]("entry") { entry =>
+      val value = implicitly[Enum[T]].withNameOption(entry)
+      Either.cond(
+        value.isDefined,
+        value.get,
+        s"Unknown enum value $entry for type ${c.runtimeClass.getName} "
+      )
+    }
+  }
+
+  implicit def oauthBearerTokenParser: core.ArgParser[OAuth2BearerToken] = {
+    core.ArgParser.instance[OAuth2BearerToken]("token") { token =>
+      //no need for left since this should never fail
+      Right(OAuth2BearerToken(token))
+    }
+  }
+
+  implicit def UUIDParser: core.ArgParser[UUID] = {
+    core.ArgParser.instance[UUID]("uuid") { uuid =>
+      Try(UUID.fromString(uuid)) match {
+        case Success(value)     => Right(value)
+        case Failure(exception) => Left(exception.getMessage)
+      }
+    }
+  }
 
-trait Command {
-  def execute(webClient: ClioWebClient, config: BaseArgs, ioUtil: IoUtil)(
-    implicit ec: ExecutionContext,
-  ): Future[HttpResponse]
+  implicit def offsetDateTimeParser: core.ArgParser[OffsetDateTime] = {
+    core.ArgParser.instance[OffsetDateTime]("date") { offsetDateAndTime =>
+      Try(OffsetDateTime.parse(offsetDateAndTime)) match {
+        case Success(value)     => Right(value)
+        case Failure(exception) => Left(exception.getMessage)
+      }
+    }
+  }
 }
+
+sealed trait CommandType
```

Comment:
Should the rest of these be moved into `Commands` to mirror the move of `CommonOptions`?

Reply:
I tried that and case-app got mad... I'll see if I can fiddle around with it more